### PR TITLE
Implement `TypedPath` for `WithRejection<TypedPath, _>`

### PR DESCRIPTION
**Pull request by davidpdrsn**
Fixes https://github.com/tokio-rs/axum/issues/2489

### CODE REVIEW

File: axum-extra/src/extract/with_rejection.rs [hunk 1]
```diff
@@ -137,6 +137,27 @@ where
     }
 }
 
+#[cfg(feature = "typed-routing")]
+const _: () = {
```
Comment (jplatte) [inline axum-extra/src/extract/with_rejection.rs:141]: Why do it this way? I find the `const _` trick a bit weird.

Comment (davidpdrsn) [reply axum-extra/src/extract/with_rejection.rs:141]: Just so you don't need to put `#[cfg(feature = "typed-routing")]` on the imports as well (unless you want warnings) or use fully qualified paths.

Comment (jplatte) [reply axum-extra/src/extract/with_rejection.rs:141]: I think I'd prefer either of those solutions 😅

cc https://github.com/rust-lang/rust/issues/65516 ;)

Comment (davidpdrsn) [reply axum-extra/src/extract/with_rejection.rs]: I hope that doesn't land 😅 I think its pretty useful to be able to define stuff everywhere though I do see its a nightmare from and IDE point of view.

But I've fixed it in c1b27565976b0

**Review by jplatte — APPROVED** [review]

**Pull request merged**